chain/chain/src/chain.rs

```diff
@@ -831,6 +832,64 @@ impl Chain {
         Ok(())
     }
 
+    pub fn apply_chunk_for_post_state_root(
+        &self,
+        shard_id: ShardId,
+        prev_state_root: StateRoot,
+        block_height: BlockHeight,
+        prev_block: &Block,
+        transactions: &[SignedTransaction],
+        last_validator_proposals: ValidatorStakeIter,
+        gas_limit: Gas,
+        last_chunk_height_included: BlockHeight,
+    ) -> Result<ApplyTransactionResult, Error> {
+        let prev_block_hash = prev_block.hash();
+        let is_first_block_with_chunk_of_version = check_if_block_is_first_with_chunk_of_version(
+            self.store(),
+            self.epoch_manager.as_ref(),
+            prev_block_hash,
+            shard_id,
+        )?;
+        // TODO(post-state-root): this misses outgoing receipts from the last block before
+        // the switch to post-state-root. Incoming receipts for that block corresponds to the
+        // outgoing receipts from the previous block, but incoming receipts for the next block
+        // include outgoing receipts for that block. These receipts can be obtained from the db
+        // using get_outgoing_receipts_for_shard since we currently track all shard. This will
+        // be implemented later along with an intergation test to reproduce the issue.
```

Comment:
Even though you explained it to me previously, it was hard to understand this by the comment. I came up with some reformulation which worked better for me - WDYT?

```suggestion
        // TODO(post-state-root): this is NOT correct for chunks ONLY for the first block B after
        // switching to post-state-root. In general case incoming receipts for post-state-root
        // chunks are derived from outgoing receipts for chunks in the previous block. However,
        // the previous block for B has only incoming receipts, as required by pre-state-root
        // logic.
        // To get incoming receipts in this edge case we can call get_outgoing_receipts_for_shard
        // which obtains receipts from DB directly. This is enough because currently each node
        // tracks all shards.
        // This will be implemented later along with an integration test to reproduce the issue.
```

Reply:
thanks for the suggestion, I've tried to make it more detailed, please let me know if that looks more understandable

Reply:
We had an offline chat. Got ideas to
- point out that we talk about `DBCol::IncomingReceipts` population and remind its definition which is ~ **receipts TO execute which node received at the block B**. So, for pre-state-root, when we receive block B, it gives us outgoing receipts from B-1 ("prev"). For post-state-root, chunk at block B stores outgoing receipts caused by itself, so they go to B as well.
- reverse direction of arrow :)